File: leet_code_75/array_string/ReverseWordsinaString/Solve.java
package leet_code_75.array_string.ReverseWordsinaString;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.StringTokenizer;

public class Solve {

    public String reverseWords(String s) {

        s.trim();

        StringTokenizer word = new StringTokenizer(s);

        List<String> res = new ArrayList<>();

        while (word.hasMoreTokens()) {

            res.add(word.nextToken());

        }

        Collections.reverse(res);

        return String.join(" ", res);

        // s.trim();
        // String[] words = s.split("\\s+");

        // StringBuilder res = new StringBuilder();

        // for (int i = words.length - 1; i >= 0; i--) {

        // res.append(words[i]);
        // if (i > 0) {
        // res.append(" ");
        // }

        // }

        // return res.toString();
    }

    public static void main(String[] args) {

        String s = "a good   example";
        Solve solve = new Solve();

        System.out.println(solve.reverseWords(s));

    }

}

File: leet_code_75/array_string/ReverseWordsinaString/solve.js
const reverseWords = (s) => {
  let words = s.split(" ");
  let res = [];

  for (let i = words.length - 1; i >= 0; i--) {
    if (words[i]) {
      res.push(words[i]);
    }
  }

  return res.join(" ");
};

const s1 = "the sky is blue";
const s2 = "  hello world  ";
const s3 = "a good   example";

console.log(reverseWords(s1));
console.log(reverseWords(s2));
console.log(reverseWords(s3));
